src/server/wesnothd/server.cpp

```diff
@@ -1650,6 +1844,44 @@ void server::handle_player_in_game(player_iterator p, simple_wml::document& data
 
 		send_to_lobby(diff);
 
+		// if this is the creation of a game from a server-side queue, need to tell all the other players in the queue to join
+		if(g.q_type() == queue_type::type::server_preset) {
+			int queue_id = g.queue_id();
+			int game_id = g.id();
+
+			if(queue_info_.count(queue_id) > 0) {
+				queue_info& info = queue_info_.at(queue_id);
+				for(const std::string& name : info.players_in_queue) {
+					auto player_ptr = player_connections_.get<name_t>().find(name);
+
+					// player is still connected and not in a game, tell them to join
+					if(player_ptr != player_connections_.get<name_t>().end() && !player_ptr->get_game()) {
+						simple_wml::document join_game_doc;
+						simple_wml::node& join_game_node = join_game_doc.root().add_child("join_game");
+						join_game_node.set_attr_int("id", game_id);
+						send_to_player(player_ptr->socket(), join_game_doc);
+					}
+
+					// remove them from any other queues they joined
+					for(int queue : player_ptr->info().get_queues()) {
+						if(queue != queue_id) {
+							queue_info& other_queue = queue_info_.at(queue);
+							if(!other_queue.players_in_queue.empty()) {
+								other_queue.players_in_queue.erase(std::remove(other_queue.players_in_queue.begin(), other_queue.players_in_queue.end(), player_ptr->info().name()));
+							}
+						}
+					}
+				}
+				// clear the queue from which the game has started
+				info.players_in_queue.clear();
```

Comment:
Players that are not needed for the created game should remain in the queue?

Reply:
Updated.